### PR TITLE
Camera Distance Lock

### PR DESCRIPTION
**Pull request by jakkeez**
The camera now respects boundaries and "zooms in" when it hits a wall or floor. 
The camera distance can be changed in playerinput header file, in a define.
Test by going into game and moving around, turning the camera and walking up to the wall to see if it clips, if it doesnt, the implementation works.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.h [hunk 1]
```diff
@@ -39,12 +39,12 @@ namespace component
 		void Update(double dt);
 		void Attack();
 		
-		void setAttackInterval(float interval);
+		void SetAttackInterval(float interval);
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.h:42]: I think this change made something in main.cpp not to compile. In main you still write this with lowerCase S. I had to comment out that line to compile the code

**Review by jocke1995 — CHANGES_REQUESTED** [review]
1. Didn't compile in main because of new name of setAttackIntervall.
2. Talk to me (Joakim), I found some bugs with the camera clipping that we can discuss when you get back to solve this 👍

**Review by Whixx — COMMENTED** [review]
Camera sensitivity "goes up" when the camera zooms.

**Review by Whixx — DISMISSED** [review]
Good work!

**Review by jocke1995 — DISMISSED** [review]
Works, just 1 small clipping which we discussed when I reviewed this. But I think its fine! gj

**Review by Whixx — APPROVED** [review]

**Review by jocke1995 — APPROVED** [review]
works good

**Pull request merged**